models/Import/Service/ResultImporter.php

```diff
@@ -0,0 +1,369 @@
+            );
+        }
+    }
+
+    /**
+     * @throws ImportResultException
+     */
+    private function updateItemOutcomeVariables(
+        AbstractRdsResultStorage $resultStorage,
+        ImportResultInput $input,
+        string $testUri,
+        float $scoreTotalMax,
+        float $updatedScoreTotal
+    ): float {
+        $deliveryExecutionUri = $input->getDeliveryExecutionId();
+
+        foreach ($input->getOutcomes() as $itemId => $outcomes) {
+            $itemUri = null;
+            $updateOutcomeVariables = [];
+            $callItemId = $this->createCallItemId($deliveryExecutionUri, $itemId);
+
+            foreach ($outcomes as $outcomeId => $outcomeValue) {
+                $outcomeVariable = $this->getItemVariable(
+                    $resultStorage,
+                    $deliveryExecutionUri,
+                    $itemId,
+                    $callItemId,
+                    $outcomeId
+                );
+
+                /** @var taoResultServer_models_classes_Variable $variable */
+                $variable = $outcomeVariable['variable'];
+                $itemUri = $outcomeVariable['itemUri'];
+                $variableId = $outcomeVariable['variableId'];
+
+                $updatedScoreTotal -= (float)$variable->getValue();
+                $updatedScoreTotal += $outcomeValue;
+
+                $variable->setValue($outcomeValue);
+
+                $updateOutcomeVariables[$variableId] = $variable;
+
+                if ($updatedScoreTotal > $scoreTotalMax) {
+                    throw new ImportResultException(
+                        sprintf(
+                            'SCORE_TOTAL_MAX cannot be higher than %s, %s provided',
+                            $scoreTotalMax,
+                            $updatedScoreTotal
+                        )
+                    );
+                }
+            }
+
+            $resultStorage->replaceItemVariables(
+                $deliveryExecutionUri,
+                $testUri,
+                $itemUri,
+                $callItemId,
+                $updateOutcomeVariables
+            );
+        }
+
+        return $updatedScoreTotal;
+    }
+
+    /**
+     * @throws ImportResultException
+     */
+    private function getTestScoreVariables(AbstractRdsResultStorage $resultStorage, string $deliveryExecutionUri): array
+    {
+        /** @var taoResultServer_models_classes_ResponseVariable $scoreTotalVariable */
+        $scoreTotalVariable = null;
+        $scoreTotal = null;
+        $scoreTotalMax = null;
+        $updatedScoreTotal = null;
+        $scoreTotalVariableId = null;
+        $outcomeVariables = $resultStorage->getDeliveryVariables($deliveryExecutionUri);
+
+        foreach ($outcomeVariables as $id => $outcomeVariable) {
+            $variable = $this->getVariable($outcomeVariable);
+
+            if ($variable === null) {
+                continue;
+            }
+
+            if ($variable->getIdentifier() === 'SCORE_TOTAL') {
```

Comment:
Maybe instead of plain text, we can store this one in a private constant?

Reply:
In this case I will create a constant with the same name and replace it here. I am not sure the benefit on it.

Reply:
You can keep plain text - for me it's not critical and will not be a blocker 🙂